### PR TITLE
split the properties class with any runtime class like SourceOrSink,

### PR DESCRIPTION
**Pull request by wwang-talend**
Source or Sink, the properties object will only return the
runtime setting object, only which is used in runtime

### CODE REVIEW

**Comment by build-talend** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/95/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/95/)

**Comment by build-talend** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/105/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/105/)

**Comment by sgandon** [comment]
I did not see that you have introduced yet another Json Library, please remove it and use one of the 2 that already exists json-io or jackson. Both of them are already available in the maven dependencies since they are both used by daikon.

File: components-jdbc/src/main/java/org/talend/components/jdbc/tjdbcconnection/TJDBCConnectionProperties.java [hunk 1]
```diff
@@ -113,8 +114,19 @@ public void refreshLayout(Form form) {
     }
 
     @Override
-    public JDBCConnectionModule getJDBCConnectionModule() {
-        return connection;
+    public AllSetting getRuntimeSetting() {
```
Comment (sgandon) [inline components-jdbc/src/main/java/org/talend/components/jdbc/tjdbcconnection/TJDBCConnectionProperties.java]: this code seems duplicated 4 times, please refactor this. Making a common class may be of help.

File: components-jdbc/src/main/java/org/talend/components/jdbc/CommonUtils.java [hunk 2]
```diff
@@ -14,4 +18,70 @@ public static Schema getSchema(SchemaProperties schema) {
     public static Form addForm(Properties props, String formName) {
         return new Form(props, formName);
     }
+
+    // get main schema from the out connector of input components
+    public static Schema getMainSchemaFromOutputConnector(ComponentProperties properties) {
```
Comment (wwang-talend) [inline components-jdbc/src/main/java/org/talend/components/jdbc/CommonUtils.java]: Yes, it should be a part of components-common module. I can move it to components-common module.

File: components-jdbc/src/main/java/org/talend/components/jdbc/dataprep/TDataPrepDBInputProperties.java [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+    protected Set<PropertyPathConnector> getAllSchemaPropertiesConnectors(boolean isOutputConnection) {
+        if (isOutputConnection) {
+            return Collections.singleton(mainConnector);
+        }
+        return Collections.emptySet();
+    }
+
+    // basic setting
+    public enum DBType {
+        MYSQL,
```
Comment (wwang-talend) [inline components-jdbc/src/main/java/org/talend/components/jdbc/dataprep/TDataPrepDBInputProperties.java]: The list ui element only provide the choice for db type, we will use the db type to load the db connection driver information(driver maven path, driver class) from a json config file. Now this is only a example for dataprep team, so no need to contain all the dbs

File: components-jdbc/src/main/java/org/talend/components/jdbc/dataprep/TDataPrepDBInputProperties.java [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+        setting.setPassword(this.userPassword.password.getValue());
+
+        setting.setSql(this.sql.getValue());
+
+        return setting;
+    }
+
+    private void setDriverAndClass(AllSetting setting) {
+        // TODO this is only a simple implement
+        InputStream is = this.getClass().getClassLoader().getResourceAsStream("db_type_config.json");
```
Comment (wwang-talend) [inline components-jdbc/src/main/java/org/talend/components/jdbc/dataprep/TDataPrepDBInputProperties.java]: Agree, in future, we will provide it, but now it's only a example now, so will fix it later.

File: components-jdbc/src/main/java/org/talend/components/jdbc/dataprep/TDataPrepDBInputProperties.java (hunk 3 above)
Comment (sgandon) [inline components-jdbc/src/main/java/org/talend/components/jdbc/dataprep/TDataPrepDBInputProperties.java]: Sorry this is not clear.
Please remove this and make the combo values dynamically loaded from the config file. This should be generic enough that only the config file is used to provide a random set of databases.

**Comment by build-talend** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/349/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/349/)

**Comment by build-talend** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/501/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/501/)

**Comment by build-talend** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/503/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/503/)

**Comment by sgandon** [comment]
LGTM, please merge.

**Pull request merged**